typescript/examples/validate/scripts/validate.ts

```diff
@@ -225,94 +224,86 @@ async function validate(
     try {
-      const buffer = new Uint8Array(DETECT_VERSION_BYTES_REQUIRED);
+      const buffer = new Uint8Array(Mcap0Constants.DETECT_VERSION_BYTES_REQUIRED);
       const readResult = await handle.read({
         buffer,
         offset: 0,
-        length: DETECT_VERSION_BYTES_REQUIRED,
+        length: Mcap0Constants.DETECT_VERSION_BYTES_REQUIRED,
       });
-      mcapVersion = detectVersion(new DataView(buffer.buffer, 0, readResult.bytesRead));
-      if (mcapVersion == undefined) {
+      isValidMcap = hasMcapPrefix(new DataView(buffer.buffer, 0, readResult.bytesRead));
+      if (!isValidMcap) {
         throw new Error(
-          `Not a valid MCAP file: unable to detect version with file header ${Array.from(buffer)
+          `Not a valid MCAP file: prefix not detected in <${Array.from(buffer)
             .map((val) => val.toString(16).padStart(2, "0"))
-            .join(" ")}`,
+            .join(" ")}>`,
         );
       }
-      log("Detected MCAP version:", mcapVersion);
+      log("MCAP prefix detected");
     } finally {
       await handle.close();
     }
   }
 
-  switch (mcapVersion) {
-    case "pre0":
-      throw new Error("MCAP pre-v0 files are not supported");
+  if (!isValidMcap) {
+    return;
+  }
 
-    case "0":
-      if (!stream) {
-        const handle = await fs.open(filePath, "r");
-        try {
-          let buffer = new ArrayBuffer(4096);
-          const reader = await Mcap0IndexedReader.Initialize({
-            readable: {
-              size: async () => BigInt((await handle.stat()).size),
-              read: async (offset, length) => {
-                if (offset > Number.MAX_SAFE_INTEGER || length > Number.MAX_SAFE_INTEGER) {
-                  throw new Error(`Read too large: offset ${offset}, length ${length}`);
-                }
-                if (length > buffer.byteLength) {
-                  buffer = new ArrayBuffer(Number(length * 2n));
-                }
-                const result = await handle.read({
-                  buffer: new DataView(buffer, 0, Number(length)),
-                  position: Number(offset),
-                });
-                if (result.bytesRead !== Number(length)) {
-                  throw new Error(
-                    `Read only ${result.bytesRead} bytes from offset ${offset}, expected ${length}`,
-                  );
-                }
-                return new Uint8Array(
-                  result.buffer.buffer,
-                  result.buffer.byteOffset,
-                  result.bytesRead,
-                );
-              },
-            },
-            decompressHandlers,
-          });
-          for (const record of reader.schemasById.values()) {
-            processRecord(record);
-          }
-          for (const record of reader.channelsById.values()) {
-            processRecord(record);
-          }
-          for await (const record of reader.readMessages()) {
-            processRecord(record);
-          }
-          break;
```

Comment:
I'm a little confused by the removed `break` — does that not have some effect (skipping over the `await readStream`)?